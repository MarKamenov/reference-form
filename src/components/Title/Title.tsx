import titleStyles from './Title.module.scss'

export interface TitleProps {
    title?: string;
}
export const Title = ({ title = 'Goodlord Referencing form' }: TitleProps) => {
    return (
        <h1 data-testid="heading" className={titleStyles.title}>{title}</h1>
    );
}
