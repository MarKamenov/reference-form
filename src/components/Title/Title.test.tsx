import { render } from '@testing-library/react';
import { screen } from '@testing-library/dom'

import { Title } from './Title';

describe('Title component', () => {

    test('should render component', () => {
        const { container } = render(<Title />)

        expect(container).toBeInTheDocument();
    });

    test('should match snapshot', () => {
        const { container } = render(<Title />);

        expect(container).toMatchSnapshot();
    });

    test('should render default title text', () => {
        render(<Title />);
        const title = 'Referencing form'
        const heading = screen.getByTestId('heading')

        expect(heading.textContent).toBe(title)
    });

    test('should render title text', () => {
        const title = 'Different title'
        render(<Title title={title} />);

        const heading = screen.getByTestId('heading')

        expect(heading.textContent).toBe(title)
    });
});

