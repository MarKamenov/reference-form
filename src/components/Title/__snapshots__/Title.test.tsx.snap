// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component should match snapshot 1`] = `
<div>
  <h1
    class="title"
    data-testid="heading"
  >
    Goodlord Referencing form
  </h1>
</div>
`;
